src/libgrate/grate-texture.c

```diff
@@ -116,30 +133,121 @@ static int grate_texture_load_internal(struct grate *grate,
 		iluScale((*tex)->pixbuf->width, (*tex)->pixbuf->height, 0);
 	}
 
+	/*
+	 * ilOriginFunc() doesn't work properly in conjunction with
+	 * ilCompressDXT(), image isn't rotated and colors are shifted.
+	 */
+	iluRotate(180.0f);
```

Comment:
It just occurred to me that DevIL is weird, it only flips vertically when 180° is used, while horizontal flip also should be done. iluRotate(179.9f) gives a correct result.

The vertical flip is exactly what is needed here, so that weirdness could be ignored. Maybe that is done on a purpose, although that odd behaviour isn't documented.